steps/optimize.py

```diff
@@ -0,0 +1,70 @@
+from zquantum.core.cost_function import get_ground_state_cost_function
+from zquantum.core.estimator import BasicEstimator
+from zquantum.core.circuit import load_circuit, load_circuit_template_params, Circuit
+from zquantum.core.utils import create_object
+from zquantum.core.serialization import save_optimization_results
+from openfermion import SymbolicOperator
+import json
+import os
+import numpy as np
+from typing import Union, Dict, Optional, List
+
+
+def optimize_parameterized_quantum_circuit(
```

Comment:
I think this name is slightly confusing – we're using `get_ground_state_cost_function` inside, and I believe that `optimize_parameterized_quantum_circuit` applies to a broader class of problems than just finding the ground state.

Reply:
Addressed here: 
14a0ec7